analyzers/src/SonarAnalyzer.Common/Rules/InsecureTemporaryFilesCreationBase.cs

```diff
@@ -0,0 +1,62 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Immutable;
+using System.Resources;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class InsecureTemporaryFilesCreationBase<TMemberAccessSyntax, TSyntaxKind> : SonarDiagnosticAnalyzer
+        where TMemberAccessSyntax : SyntaxNode
+        where TSyntaxKind : struct
+    {
+        protected const string DiagnosticId = "S5445";
+        private const string VulnerableApiName = "GetTempFileName";
+        private const string MessageFormat = "'Path.GetTempFileName()' is insecure. Use 'Path.GetRandomFileName()' instead.";
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        private DiagnosticDescriptor Rule { get; }
+
+        protected InsecureTemporaryFilesCreationBase(ResourceManager rspecResources) =>
+            Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, rspecResources);
+
+        protected abstract TSyntaxKind SyntaxKind { get; }
```

Comment:
GeneratedCodeRecognizer and SyntaxKind can be replaced by protected abstract ILanguageFacade<TSyntaxKind> Language { get; }.  It has the coderecognizer, and it has a SyntaxFacade that can be extended with a SimpleMemberAccessExpression and can be used directly in the base class.